engine/src/python/config/python_utils.cpp

```diff
@@ -65,18 +64,51 @@ PyObject* GetClassFromPython(
         // TODO: throw exception
         return nullptr;
     }
-    
+
+    PyPreConfig py_pre_config;
+    PyPreConfig_InitPythonConfig(&py_pre_config);
+
+    PyStatus status;
+
+    status = Py_PreInitialize(&py_pre_config);
+    if (PyStatus_Exception(status)) {
+        Py_ExitStatusException(status);
+    }
+
+    const std::string python_path = GetPythonPath();
+    const std::wstring python_path_w(python_path.begin(), python_path.end());
+    const std::wstring path_string_w = std::wstring(path_string.begin(), path_string.end());
     const std::string yaml_path = QUOTE(Python_SITELIB);
-    const std::string python_path_string = GetPythonPath() 
-                                           + ":" + path_string
-                                           + ":" + yaml_path + ":" + build_path;
-    const std::wstring python_path_wstring = std::wstring(python_path_string.begin(), 
-                                                          python_path_string.end());
-    const wchar_t* python_path = python_path_wstring.c_str();
-    Py_SetPath(python_path);
-    Py_Initialize();
+    const std::wstring yaml_path_wstring = std::wstring(yaml_path.begin(), yaml_path.end());
+    const std::wstring build_path_w = std::wstring(build_path.begin(), build_path.end());
+
+    PyWideStringList python_path_py_wide_string_list{};
+    status = PyWideStringList_Append(&python_path_py_wide_string_list, python_path_w.c_str());
+    if (PyStatus_Exception(status)) {
+        Py_ExitStatusException(status);
+    }
+    status = PyWideStringList_Append(&python_path_py_wide_string_list, path_string_w.c_str());
+    if (PyStatus_Exception(status)) {
+        Py_ExitStatusException(status);
+    }
+    status = PyWideStringList_Append(&python_path_py_wide_string_list, yaml_path_wstring.c_str());
+    if (PyStatus_Exception(status)) {
+        Py_ExitStatusException(status);
+    }
+    status = PyWideStringList_Append(&python_path_py_wide_string_list, build_path_w.c_str());
+
+    PyConfig config;
+    PyConfig_InitPythonConfig(&config);
+
+    config.module_search_paths = python_path_py_wide_string_list;
+    config.isolated = 0;
+
+    status = Py_InitializeFromConfig(&config);
```

Comment:
This looks much better than the code I cobbled together. One thing I had and lost as I started working on python is the ability to change the python file and se the changes in RT. That would be great to get back. Not sure what I did to break this.

Reply:
Glad you like it :-)